File: src/ru/hudspeed/pro/RadarBaseUpdateSingleFlight.java
package ru.gpsantiradar.app;

import java.util.concurrent.atomic.AtomicBoolean;

final class RadarBaseUpdateSingleFlight {
    private final AtomicBoolean running = new AtomicBoolean();

    boolean tryStart() {
        return running.compareAndSet(false, true);
    }

    void finish() {
        running.set(false);
    }
}

File: src/main/kotlin/ru/gpsantiradar/app/RadarBaseUpdateSingleFlight.kt
package ru.gpsantiradar.app

import java.util.concurrent.atomic.AtomicBoolean

class RadarBaseUpdateSingleFlight {
    private val running = AtomicBoolean()
    fun tryStart() = running.compareAndSet(false, true)
    fun finish() = running.set(false)
}
